### ISSUE TITLE
red5 live stream: no video with flowplayer-3.2.15.swf (regression)

### ISSUE THREAD

**Issue by phloxic** [flowplayer/flash#28]
The following config works in 3.2.14, but with 3.2.15 there is no video:

``` js
plugins: {
  rtmp: {
    url: 'flowplayer.rtmp-3.2.11.swf',
    netConnectionUrl: 'rtmp://tsingyuan.cn/oflaDemo'
  },
  controls: {
    scrubber: false
  }
},
clip: {
  provider: 'rtmp',
  url: '803428',
  live: true
}
```

See http://flowplayer.org/forum/8/103485 - 3.2.14 works in http://flowplayer.org/forum/8/103485#post-103561

red5-1.0.0-RC2

**Comment by danrossi** [comment]
Hi it's likely because no metadata /  start event is sent therefore the screen display is not added. I don't know what to do about that because it will go back and messup stage video. Are all begin /metadata / start events being sent ? Basically if we go back to some code being called for each bufferfull event hence why it seems to be working in the previous versions  it will mess up other things.

**Comment by phloxic** [comment]
It must be something regarding red5 as this demo works with 3.2.15:
http://flowplayer.blacktrash.org/test/live.html

**Comment by danrossi** [comment]
ok ill give it a test thanks.

**Comment by phloxic** [comment]
http://flowplayer.org/forum/8/103485#post-103647 :

_In fact it is not only related to red5/rtmp as there is the same problem with http delivered live video from Bitgravity CDN._
_It seems that there is no metadata when the stream starts playing. Setting 'metaData:false' does not change anything._

_In my case there is no rtmp plugin at all. The live stream works fine with 3.2.14._

**Comment by danrossi** [comment]
OK your red5 test shows correctly. So live streams with no metadata will therefore not dispatch onstart. perhaps the code needs to move to onbegin which is when the play event is sent. start is sent one metadata is recieved ???

**Comment by danrossi** [comment]
ok if start doesnt get dispatched at all it will do this because of the changes to the display adding, ill have to rework this yet again to happen on the begin event. I added the code from within the start listener to in the begin listener and seems to work fine if no metadata is triggered. 

There is alot of duplicated code around the screen display stuff and ive tried my best to make it work for all possibly solutions. 

https://dl.dropbox.com/u/3394987/flowplayer.commercial-3.2.15.zip

If you know something that doesnt have metadata please try it. 

basically begin is triggered on the flash stream play event, start is triggered on the metadata callback.no metadata no onStart triggered :(

**Comment by phloxic** [comment]
I can't really comment, as I have no experience with streams lacking metadata. imho it's ok to not support events depending on metadata when there is no metadata. On the other hand how do you determine onStart in other situations, e.g. playback starts? Must be happening after onMetaData.

**Comment by danrossi** [comment]
its only triggered once metdata is received yes. there is quite a few things that rely on start events i think also, so i think there should be the requirement of metdata not sure. if that file works for you ill have a look at cleaning things up more.

**Comment by phloxic** [comment]
Just try it with the originally reported config - but it might not be live.
Then it's best to take this up here:
http://flash.flowplayer.org/forum/8/103485

**Comment by danrossi** [comment]
fixed in https://dl.dropboxusercontent.com/u/3394987/flowplayer.rtmp-3.2.12.zip. It will set a timeout and wait for metadata on live streams and if none available trigger a start event. red5 triggers metadata on h264 encoding not vp6. They know about it now.

**Comment by danrossi** [comment]
closing this for now.
